src/state/_redux/user/middleware.ts

```diff
@@ -0,0 +1,65 @@
+import { Middleware } from 'redux';
+import { getType } from 'typesafe-actions';
+import axios from 'axios';
+
+import { IStore } from '../../../boot/configureStore';
+import { createUser, loginUser } from '@state/_redux/user/actions';
+import { setJwt } from '@utils/jwt';
+
+const creatUserRequest = async (action: any, dispatch: any) => {
+    const { email, firstName, lastName, password } = action.payload;
+    try {
+        const response = await axios.post('http://localhost:8084/auth/user', {
+            email,
+            password,
+            firstName,
+            surname: lastName,
+        });
+        dispatch(createUser.success(response));
+        return true;
+    } catch (err) {
+        dispatch(createUser.failure(err));
+        return false;
+    }
+};
+
+const loginRequest = async (action: any, dispatch: any) => {
```

Comment:
import { Dispatch } from 'redux';  dispatch : Dispatch type. for action import proper action I think